scripts/west_commands/patch.py

```diff
@@ -199,18 +253,23 @@ def filter_args(self, args):
 
     def do_run(self, args, _):
         self.filter_args(args)
-
-        if not os.path.isfile(args.patch_yml):
-            self.inf(f"no patches to apply: {args.patch_yml} not found")
-            return
+        yml = None
 
         west_config = Path(args.west_workspace) / ".west" / "config"
         if not os.path.isfile(west_config):
             self.die(f"{args.west_workspace} is not a valid west workspace")
 
+        if not os.path.isfile(args.patch_yml):
+            if args.subcommand in ["gh-fetch"]:
+                yml = {"patches": []}
+            else:
+                self.inf(f"no patches to apply: {args.patch_yml} not found")
+                return
+
         try:
-            with open(args.patch_yml) as f:
-                yml = yaml.load(f, Loader=SafeLoader)
+            if not yml:
+                with open(args.patch_yml) as f:
+                    yml = yaml.load(f, Loader=SafeLoader)
             if not yml:
                 self.inf(f"{args.patch_yml} is empty")
                 return
```

Comment:
I kind of wonder if this section (and some lines beneath it) should be converted to a (possibly static) method, with a bool specifying whether non-existent / empty is ok.

Reply:
That's fair, I'll look into this.